src/Monotonic.cpp

```diff
@@ -493,10 +347,9 @@ class DerivativeBounds : public IRVisitor {
             return;
         }
 
-        if (op->is_intrinsic(Call::unsafe_promise_clamped) ||
-            op->is_intrinsic(Call::promise_clamped) ||
-            op->is_intrinsic(Call::saturating_cast)) {
+        if (op->is_intrinsic(Call::saturating_cast)) {
```

Comment:
Why the removal of `(unsafe_)promise_clamped`?

Reply:
Because derivative_bounds calls remove_likelies and remove_promises before starting the visitor